features/nvstore/source/nvstore.cpp

```diff
@@ -550,40 +550,27 @@ int NVStore::do_get(uint16_t key, uint16_t buf_len_bytes, uint32_t *buf, uint16_
         return NVSTORE_BUFF_NOT_ALIGNED;
     }
 
-    // This loop is required for the case we try to perform reading while GC is in progress.
-    // If so, we have the following cases:
-    // 1. Record is still in the older area. It will be successfully read.
-    // 2. Record was already copied to the new area. Now the _offset_by_key indicates it.
-    // So we have two cases here:
-    // a. Read from new area succeeds. Everything's OK.
-    // b. Read fails (either physically or CRC error). So we know that if the area taken
-    //    from _offset_by_key is different from the active area, a GC is in progress, so
-    //    retry the operation.
-    for (;;) {
-        record_offset = _offset_by_key[key];
-        if (!record_offset) {
-            return NVSTORE_NOT_FOUND;
-        }
+    // We only have issues if we read during GC, so shared lock is required.
+    _write_lock.shared_lock();
```

Comment:
Yes, but shouldn't really be called `write_lock` any more. It's I guess it's a "layout" lock? Or a "gc" lock?

Reply:
Hmmm... Maybe just lock...